platform-workflow-module/src/main/java/com/softicar/platform/workflow/module/workflow/transients/field/WorkflowUserField.java

```diff
@@ -61,22 +63,41 @@ private List<String> getUserTaskAgeStrings(R row) {
 			.collect(Collectors.toList());
 	}
 
-	private Map<AGUser, DayTime> getUserTaskCreationMap(Set<AGWorkflowTask> tasks) {
+	private Map<Pair<AGUser, AGUser>, DayTime> getUserTaskCreationMap(Set<AGWorkflowTask> tasks) {
 
 		AGWorkflowTask.CREATION_TRANSACTION.prefetchAll(tasks);
 
-		TreeMap<AGUser, DayTime> userTaskCreationMap = new TreeMap<>();
+		Map<Pair<AGUser, AGUser>, DayTime> userTaskCreationMap = new HashMap<>();
 		for (AGWorkflowTask task: tasks) {
 			Optional<AGTransaction> creationTransaction = Optional.ofNullable(AGWorkflowTask.CREATION_TRANSACTION.getValue(task));
-			if (creationTransaction.isPresent()) {
-				userTaskCreationMap.merge(task.getUser(), creationTransaction.get().getAt(), DayTime::min);
+
+			if (creationTransaction.isPresent() && !task.hasActiveDelegation()) {
+				userTaskCreationMap.merge(new Pair<>(task.getUser(), task.getUser()), creationTransaction.get().getAt(), DayTime::min);
+			} else if (task.hasActiveDelegation()) {
+				AGWorkflowTaskDelegation delegation = task.getDelegation().get();
+				Optional<DayTime> lastModificationTimeFormLog = delegation.getLastModificationTimeFormLog();
+				if (lastModificationTimeFormLog.isPresent()) {
+					userTaskCreationMap.merge(new Pair<>(task.getUser(), delegation.getTargetUser()), lastModificationTimeFormLog.get(), DayTime::min);
+				}
 			}
+
 		}
 		return userTaskCreationMap;
 	}
 
-	private String getUserTaskAgeString(AGUser user, int days) {
+	private String getUserTaskAgeString(Pair<AGUser, AGUser> userAndCurrentDelegation, int days) {
```

Comment:
I think, instead of using `Pair<AGUser, AGUser>` all over the place, we should create a private class with properly-named fields and getters. That should add some clarity.

Reply:
Added as private inner class, please check if this is OK.